MCH: added option to output the time-clustered ROFs without filtering

aferrero2707 opened the pull request:
The additional output containing all the time-clustered ROFs without futher filtering can be useful for diagnistics, oir example in online QC. 
The output is disabled by default, and can be enabled by setting `"MCHTimeClusterizer.enableUnfilteredOutput=true"` in the configuration keys.
The default name of the DPL message is `"UTC-F-DIGITROFS"`. It can be customized via the `--output-unfiltered-digitrofs-data-description` command-line option.

aferrero2707 commented:
@aphecetche this PR introduces the possibility to output the time-clustered ROFs without any filtering, something we discussed in the past and that could be useful for online QC. Could you have a look? Thanks!

alibuild commented:
Error while checking build/O2/fullCI for 0598627cbaf8937a543dbdee6ab36421e014c4d2 at 2023-08-21 04:53:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2Physics-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[0 more errors; see full log]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/11762/0598627cbaf8937a543dbdee6ab36421e014c4d2/build_O2_fullCI/pretty.html).

pillot commented on Detectors/MUON/MCH/TimeClustering/include/MCHTimeClustering/TimeClusterFinderSpec.h:
```diff
@@ -28,6 +28,8 @@ o2::framework::DataProcessorSpec
                            std::string_view inputDigitDataDescription = "F-DIGITS",
                            std::string_view inputDigitRofDataDescription = "F-DIGITROFS",
+                           bool enableUnfilteredOutput = false,
+                           std::string_view outputUnfilteredDigitRofDataDescription = "UTC-F-DIGITROFS",
```
I would put the new arguments after "outputDigitRofDataDescription", which is set in reco-workflow.cxx and would now be mixed up with "enableUnfilteredOutput", unless you also change reco-workflow.cxx accordingly.

aferrero2707 replied:
Good point! I have done as you suggested, without changing reco-workflow.cxx

pillot commented on Detectors/MUON/MCH/TimeClustering/src/digits-to-timeclusters-workflow.cxx:
```diff
@@ -39,11 +41,14 @@ void customize(std::vector<ConfigParamSpec>& workflowOptions)
 WorkflowSpec defineDataProcessing(const ConfigContext& cc)
 {
   o2::conf::ConfigurableParam::updateFromString(cc.options().get<std::string>("configKeyValues"));
+  const auto& param = o2::mch::TimeClusterizerParam::Instance();
```
I would check this configurable parameter directly inside getTimeClusterFinderSpec() instead of passing it as an argument. This way the saving of the extra ouput would be activable for every workflows that use this spec (i.e. from reco-workflow too) without having to modify each of them to catch this parameter.

aferrero2707 replied:
Indeed, done as suggested.

pillot reviewed (commented):
Laurent will make the final review but in the meantime I have a couple of suggestions if it may help.
Not sure this is what you want but currently this extra output cannot be enabled from the reco-workflow which is used in online reco. Beside the current changes will alter this workflow (see detailed comments).

aphecetche commented:
I do agree with Philippe comments.

aferrero2707 committed:
[MCH] added option to output the time-clustered ROFs without filtering

The additional output containing all the time-clustered ROFs without futher filtering can be useful for diagnostics, for example in online QC.
The output is disabled by default, and can be enabled by setting `"MCHTimeClusterizer.enableUnfilteredOutput=true"` in the configuration keys.
The default name of the DPL message is `"UTC-F-DIGITROFS"`. It can be customized via the `--output-unfiltered-digitrofs-data-description` command-line option.

aferrero2707 commented:
@ktf there are two CI failures that seem to be un-related to this PR. Would it be possible to re-run those CI jobs? Or should I just push some dummy commit to re-trigger the CI? Thanks!
